--- a/ecm_conntrack_notifier.c
+++ b/ecm_conntrack_notifier.c
@@ -332,8 +332,14 @@
  * ecm_conntrack_event()
  *	Callback event invoked when conntrack connection state changes, currently we handle destroy events to quickly release state
  */
+#ifdef CONFIG_NF_CONNTRACK_CHAIN_EVENTS
+static int ecm_conntrack_event(struct notifier_block *this, unsigned long events, void *ptr)
+{
+	struct nf_ct_event *item = (struct nf_ct_event *)ptr;
+#else
 static int ecm_conntrack_event(unsigned int events, const struct nf_ct_event *item)
 {
+#endif
 	struct nf_conn *ct = item->ct;
 
 	/*
@@ -387,6 +393,11 @@
  * struct nf_ct_event_notifier ecm_conntrack_notifier
  *	Netfilter conntrack event system to monitor connection tracking changes
  */
+#ifdef CONFIG_NF_CONNTRACK_CHAIN_EVENTS
+static struct notifier_block ecm_conntrack_notifier = {
+	.notifier_call	= ecm_conntrack_event,
+};
+#else
 static struct nf_ct_event_notifier ecm_conntrack_notifier = {
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0))
     .fcn	= ecm_conntrack_event,
@@ -394,6 +405,7 @@
 #endif
 };
 #endif
+#endif
 
 /*
  * ecm_conntrack_notifier_stop()
@@ -430,7 +442,7 @@
 	/*
 	 * Eventing subsystem is available so we register a notifier hook to get fast notifications of expired connections
 	 */
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0))
+#if defined(CONFIG_NF_CONNTRACK_CHAIN_EVENTS) || (LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0))
    result = nf_conntrack_register_notifier(&init_net, &ecm_conntrack_notifier);
 	if (result < 0) {
 		DEBUG_ERROR("Can't register nf notifier hook.\n");
@@ -481,7 +493,7 @@
 #ifdef ECM_IPV6_ENABLE
 	nf_ct_netns_put(&init_net, NFPROTO_IPV6);
 #endif
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0))
+#if defined(CONFIG_NF_CONNTRACK_CHAIN_EVENTS) || (LINUX_VERSION_CODE < KERNEL_VERSION(5, 15, 0))
 	nf_conntrack_unregister_notifier(&init_net, &ecm_conntrack_notifier);
 #else
    nf_conntrack_unregister_notifier(&init_net);
